APID: Doc changes for 2.13.0 (#133)
mulesoft/docs-design-center: modules/ROOT/pages/design-publish.adoc

= Publish an API Specification
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]

:imagesdir: ../assets/images

> When you want to share your API specification with other developers, you can publish it to Anypoint Exchange. Exchange creates a page that presents the project for anyone in your business group to view and download the project from.

Reviewer: colon?

.The page that Exchange creates for a published project includes a left pane for viewing the specification's documentation, a middle pane for viewing a description of the specification, and a right pane for viewing metadata about the specification.
image:apid-example-page-in-exchange.png[An example page for a specification published in Exchange]

<1> The left pane, which you can navigate to view the documentation. The navigation is generated from the documentation in the specification.
<2> The middle pane, which describes the specification. You write the description in Exchange after publishing the specification.
<3> The right pane, which displays metadata about the specification.

> == Procedure

Reviewer 2: We should explain that one of the steps is to `set the root-file`.
For projects that were created before this release, it is possible that the root-file is not set. In these cases when the Project is opened the following message is displayed:
![2020-01-31_11-16-05](https://user-images.githubusercontent.com/12468128/73546151-779ff980-441b-11ea-9c89-eefbc4797bd5.png)

If the root-file is not set, it cannot be published to Exchange.

Author: @iamerico , I've added this note to the doc about root files:

> The first time that you open in the text editor a project that was created before the February 8, 2020 release of API Designer, you are prompted to set a root file.

. Open the API-specification project that you want to publish, if it is not already open.

> . In the top-right corner of API Designer, click *Publish* and then click *Publish to Exchange*.

Reviewer: ```suggestion
. In the top-right corner of API Designer, click *Publish* and then click *Publish to Exchange*:
```

image:apid-publish-dialog-api-specs.png[]

> 

Reviewer: For the following list (and all similar callout lists), I suggest breaking the item into a new line after the initial description fragment or sentence (and attempting to make any one group like this either all fragments or all sentences:
<1> *Asset version*: The version number to be displayed in the *Version* section of the page that Anypoint Exchange creates for the project
<2> *API version*: The version number of the API specification
For a description of this field, see the next section.

Author: @kbostrom , Asciidoc callouts don't allow that, unfortunately.

<1> *Asset version*: The version number to be displayed in the *Version* section of the page that Anypoint Exchange creates for the project.

> <2> *API version*: The version number of the API specification. For a description of this field, see the next section.

Reviewer: Please name the section rather than referring to it as "the next section," to help keep topics independent organization. Just a good habit to prepare for potential reuse or reorganization.

<3> *Business group ID*: The ID of the business unit in Anypoint Platform to which the project belongs.
<4> *Asset name*: The name that Anypoint Exchange uses to identify the published project.
<5> *Asset ID*: The ID that Anypoint Exchange uses to identify the published project.

> <6> *Add to API Manager*: Select this option if you want to publish the API specification to an API portal. When you click *Publish to Exchange*, API Designer also publishes the specification to API Manager, where the specification is listed in the *API Administration* section.

Reviewer: for parallelism, consider
```suggestion
<6> *Add to API Manager*: Specifies that you want to publish the API specification to an API portal. When you click *Publish to Exchange*, API Designer also publishes the specification to API Manager, where the specification is listed in the *API Administration* section.
```

> == The *API version* field

Reviewer: When I see a title like this, I kind of wonder "What _about_ <this named thing>?"
Is the text going to tell me what this thing is? How this thing works? Statistics about or components?
Is it a task, a concept, or reference?
Just something to think about going forward.

The appearance of the *API version* field depends upon several factors:

* Anypoint Exchange requires that each API specification published to it have a version number.
* API Manager requires that API specifications provide a version number.
* In specifications written in OAS, the `version` field is required.
* In specifications written in RAML, the `version` node is optional.

> The following flowchart shows how API Designer decides whether to disable or enable the *API version* field and whether to populate it with a number.

Reviewer: ```suggestion
The following flowchart shows how API Designer decides whether to disable or enable the *API version* field and whether to populate it with a number:
```

.The appearance of the *API version* field depends on the outcome of the checks that API Designer makes when you want to publish an API specification.
image:apid-flowchart-api-version.png[Flowchart for the checks that API Designer makes]

> <1> API Designer checks whether the version number for the API specification is provided within the specification. If the version number is provided, then the *API version* field is populated with the number and disabled, so that you cannot change the number. If the specification does not provide the version number, then API Designer moves to the next check.

Reviewer: GLOBAL for this list:
```suggestion
<1> API Designer checks whether the version number for the API specification is provided within the specification. 
If the version number is provided, then the *API version* field is populated with the number and disabled, so that you cannot change the number. If the specification does not provide the version number, then API Designer moves to the next check.
```

Author: @kbostrom , can't do this with Asciidoc callouts.

> <2> API Designer checks whether the check box *Add to API Manager* is selected. If the check box is selected, then API Designer prevents you from proceeding because API Manager requires that the specification provide the version number. If the check box is not selected, then API Designer moves to the last check.

Reviewer: Our term presentation list standardizes to "checkbox" as one word
```suggestion
<2> API Designer checks whether the checkbox *Add to API Manager* is selected. If the checkbox is selected, then API Designer prevents you from proceeding because API Manager requires that the specification provide the version number. If the checkbox is not selected, then API Designer moves to the last check.
```
